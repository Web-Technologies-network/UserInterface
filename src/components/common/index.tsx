export { Card } from './Card/Card';
export { Danger } from './Typography/Danger';
export { Info } from './Typography/info';
export { Success } from './Typography/Success';
export { Warning } from './Typography/Warning';
