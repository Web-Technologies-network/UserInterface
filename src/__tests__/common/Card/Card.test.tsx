import React from 'react';
import renderer from 'react-test-renderer';
import { Card } from '../../../components/common';

/**
 * Test suite
 */
describe('Card:', () => {
  /**
   * Test case
   */
  test('should render the same', () => {
    //prepare
    const children = <div>Test children</div>;
    //act
    const tree = renderer.create(
      <Card chart={<div>Chart</div>} className={null} plain={null} profile={null}>
        {children}
      </Card>,
    );
    //assert
    expect(tree.toJSON()).toMatchInlineSnapshot(`
      <div
        className="MuiGrid-root makeStyles-card-1 makeStyles-cardChart-4 null"
      >
        <div>
          Test children
        </div>
      </div>
    `);
  });
});
