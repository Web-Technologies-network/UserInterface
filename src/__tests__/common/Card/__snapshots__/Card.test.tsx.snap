// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card: should render the same 1`] = `
<div
  className="MuiGrid-root makeStyles-card-1 makeStyles-cardChart-4 null"
>
  <div>
    Test children
  </div>
</div>
`;
